glean-core/src/histogram/functional.rs

```diff
@@ -34,11 +38,13 @@ impl Functional {
     /// mathematical concept, even though the internal representation is stored and
     /// sent using the minimum value in each bucket.
     fn sample_to_bucket_index(&self, sample: u64) -> u64 {
+        let _fpc = FloatingPointContext::new();
```

Comment:
Hm, reading this again, I wonder if we should explicitly call out that we rely on the `Drop` behavior to reset the behavior?
In Rust _not_ calling Drop is safe. The way we use it it's guaranteed it's called, but we might call it out.
The alternatives would be:

* Wrap it in a closure, that way we set and re-set on entering and exiting.
* or explicitly call `mem::drop(fpc)` to make it clear.

I don't want to block on this, it's safe behavior for now.

Reply:
I'm happy to clean this up to make it more obvious / safe.

Just curious though -- where is it documented that `Drop` might not be called?  The [Drop docs](https://doc.rust-lang.org/std/ops/trait.Drop.html) and the [Drop chapter](https://doc.rust-lang.org/book/ch15-03-drop.html) say it runs when the variable goes out of scope.  The goal here is certainly to minimize the scope to just around the floating point operation, and I wouldn't want this to have effect any longer.

Reply:
Sorry, should have been more specific:

Drop _is_ called at the end of the scope, that part is guaranteed, especially in our case.

However, in general, it _is_ safe* as per language definition for `Drop` not to be called (it can happen through reference-count cycles, explicit `mem::forget`s or other things). Lots of stuff relies on drop for correct behavior (e.g. unlocking a mutex on drop), but not for safe behavior.

(\* safe here is the Rust definition of safety as "memory-safety")

I just want to save ourselves from forgetting what we rely on in this piece of code when we read it again in 6 weeks.

Reply:
Ah, that makes sense.  I think maybe the best course of action is to comment all of these call sites -- at a minimum it's not clear why this unused variable is there and someone might "remove the cruft" (they would break the unit tests on Windows x86 that we don't yet have in CI).